### PR TITLE
Create pdf-grammar.txt

### PR DESCRIPTION
**Pull request by lwli11**
These requests add the ABNF pdf grammar and wrapper lisp function that parses and checks that the rules are closed.

### CODE REVIEW

**Comment by ericwhitmansmith** [comment]
Please change to target the acl2:testing branch instead.  See https://www.cs.utexas.edu/users/moore/acl2/manuals/latest/?topic=ACL2____GITHUB-COMMIT-CODE-USING-PULL-REQUESTS

**Comment by lwli11** [comment]
Ok, I'm trying again. I don't know where the infinite files came from.

**Comment by MattKaufmann** [comment]
This needs to go to testing instead of master.  It appears still to be targeted to master.  The "infinite files" were for a different pull request from you (#1522), by the way.

File: books/kestrel/abnf/examples/pdf-grammar.txt [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+dict-key = backslash *char ;
+
+dict-value = boolean / number / name / array / dictionary / null / rectange / reference / string ;
+
+start-dict = %x3c.3c        ; <<<
+
+end-dict = %x3e.3e          ; >>
+
+ws = *(
+
```
Comment (acoglio) [inline books/kestrel/abnf/examples/pdf-grammar.txt:74]: There's an extra blank line and the `%x00` is further right than the other codes. Can the layout be made more uniform?

**Review by acoglio — CHANGES_REQUESTED** [review]
Thank you so much for this addition. Would you mind remind this file to `pdf.abnf`, or consistency with the other grammar files in this directory? (One benefit of using an `.abnf` extension is that GitHub shows ABNF syntax highlighting.)

**Comment by lwli11** [comment]
Thanks for working with me to merge all these changes! I've renamed the changes and figured out how to add the wrapper pdf.lisp to this merge request as well.

**Review by acoglio — APPROVED** [review]

**Pull request merged**